libsolidity/formal/SMTEncoder.cpp

```diff
@@ -735,6 +746,26 @@ void SMTEncoder::visitGasLeft(FunctionCall const& _funCall)
 		m_context.addAssertion(symbolicVar->currentValue() <= symbolicVar->valueAtIndex(index - 1));
 }
 
+void SMTEncoder::visitObjectCreation(FunctionCall const& _funCall)
+{
+	auto const& args = _funCall.arguments();
+	solAssert(args.size() >= 1, "");
+	auto argType = args.front()->annotation().type->category();
+	solAssert(argType == Type::Category::Integer || argType == Type::Category::RationalNumber, "");
+
+	// TODO: support the special case of integer literals to have less symbolic variables, but this works too
```

Comment:
I think it's fine like this, don't think it's gonna help much to have it separate.

Reply:
It is actually 3 more lines of code, but we should consider that if solving is becoming too slow and we have enough heavy examples.

Reply:
But why would it be easier in that case?

Reply:
It is two useless assertions less (`setSymbolicUnknownValue` adds a min/max).

Reply:
ah right

Reply:
We could also add an optimizer step removing such assertions :wink: